src/controllers/FacetQueryController.ts

```diff
@@ -312,13 +305,26 @@ export class FacetQueryController {
   }
 
   private checkForFacetSearchValuesToRemove(fieldValues: IIndexFieldValue[], valueToCheckAgainst: string) {
-    let regex = FacetUtils.getRegexToUseForFacetSearch(valueToCheckAgainst, this.facet.options.facetSearchIgnoreAccents);
+    const regex = FacetUtils.getRegexToUseForFacetSearch(valueToCheckAgainst, this.facet.options.facetSearchIgnoreAccents);
 
     return _.filter<IIndexFieldValue>(fieldValues, fieldValue => {
-      let isAllowed = _.isEmpty(this.facet.options.allowedValues) || _.contains(this.facet.options.allowedValues, fieldValue.value);
-
-      let value = this.facet.getValueCaption(fieldValue);
+      const isAllowed = _.isEmpty(this.facet.options.allowedValues) || this.isValueAllowedByAllowedValueOption(fieldValue);
+      const value = this.facet.getValueCaption(fieldValue);
       return isAllowed && regex.test(value);
     });
   }
+
+  private filterByAllowedValueOption(values: IGroupByValue[]): IGroupByValue[] {
+    return _.filter(values, (value: IGroupByValue) => this.isValueAllowedByAllowedValueOption(value));
+  }
+
+  private isValueAllowedByAllowedValueOption(value: IGroupByValue | IIndexFieldValue): boolean {
+    // Allowed value option on the facet should support * (wildcard searches)
+    // We need to filter values client side the index will completeWithStandardValues
+    // Replace the wildcard (*) for a regex match (.*)
+    return _.some(this.facet.options.allowedValues, allowedValue => {
+      const regex = new RegExp(`^${allowedValue.replace(/\*/g, '.*').replace(/\?/g, '.')}$`, 'gi');
```

Comment:
Why do we have to replace the question mark (?) ? No test for it either.

Reply:
The index support it.

I'll add a test for it, and improve the comment.